File: App.Models/DTOs/CreateRequests/DiscountCreateRequest.cs
using System;
using App.Common.Model.DTOs;

namespace App.Models.DTOs.CreateRequests
{
    public class DiscountCreateRequest : BaseCreateRequest
    {
        public int? PercentDiscount { get; set; }
        public double? MaxDiscountPrice { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
    }
}

File: admin-angular/src/app/models/create-requests/discount-create-request.ts
import { AutoMap } from "@automapper/classes";
import { BaseCreateRequest } from "..";

export class DiscountCreateRequest extends BaseCreateRequest{
    @AutoMap()
    percentDiscount: number
    @AutoMap()
    maxDiscountPrice: number
    @AutoMap()
    fromDate: Date
    @AutoMap()
    toDate: Date
}
